Express-API/express.js

```diff
@@ -1,105 +1,82 @@
 const cors = require('cors')
 const express = require('express')
-var path = require('path')
 
 const app = express()
 const PORT = 8000
 
 const Database = require('./db.js')
 const LoginHandler = require('./login.js')
 const RolesHandler = require('./roles.js')
-const CapabilityHandler = require('./capability')
+const CapabilityHandler = require('./capability.js')
+const BandHandler = require('./band.js')
+const DepartmentHandler = require('./department.js')
 const db = new Database()
 const loginHandler = new LoginHandler()
 const rolesHandler = new RolesHandler()
 const capabilityHandler = new CapabilityHandler()
+const bandHandler = new BandHandler()
+const departmentHandler = new DepartmentHandler()
 
 // Parse URL-encoded bodies (as sent by HTML forms)
 app.use(express.urlencoded())
-
 //Parse JSON bodies (as sent by API clients)
 app.use(express.json())
 
-const rolesQuery = "SELECT role_id, name, department_id, band_id FROM role WHERE department_id = ?"
-const capabilitiesQuery = "SELECT name, capability_id FROM capability WHERE department_id = ?"
-const departmentsQuery = "SELECT name FROM department WHERE department_id = ?"
-const rolesInDepQuery = "SELECT * FROM viewTableData WHERE department_id = ?"
-const bandsQuery = "SELECT * FROM band"
-const getRoleQuery = "SELECT role_id,name,summary,job_spec_url,school_id FROM viewRoleData WHERE role_id = ?"
-
-const tokenCookieName = 'token'
+const tokenCookieName = "token"
 
-app.post('/addrole', cors(), function (request, response) {
-	rolesHandler.createRole(request.body, db).then(result => {
+function sendResponseData(query, response){
+  query.then(result => {
     response.send(result)
   }, reject => {
     response.send(reject)
   })
-});
+}
 
-app.post('/addcapability', cors(), function (request, response) {
-  capabilityHandler.createCapability(request.body, db).then(result => {
-    response.send(result)
-  }, reject => {
-    response.send(reject)
-  })
-});
+app.post("/addrole", cors(), function (request, response) {
+	sendResponseData(rolesHandler.createRole(request.body, db), response)
+})
+
+app.post("/addcapability", cors(), function (request, response) {
+  sendResponseData(capabilityHandler.createCapability(request.body, db), response)
+})
 
 app.get("/roles", cors(), function(request, response) {
-  db.query(rolesQuery,[request.query.departmentID]).then(rows => {
-    response.send(rows);
-  });
-});
+  sendResponseData(rolesHandler.getRolesInDept(request.query, db), response)
+})
 
 app.get("/capabilities", cors(), function(request, response) {
-  db.query(capabilitiesQuery,[request.query.departmentID]).then(rows => {
-    response.send(rows);
-  });
-});
+  sendResponseData(capabilityHandler.getCapabiltiies(request.query, db), response)
+})
```

Comment:
nice job refactoring this, looking a lot tidier

Reply:
cheers